src/AutoRest.CSharp/Common/Output/Models/Types/SignatureType.cs

```diff
@@ -0,0 +1,318 @@
+                if (previousMethod.ReturnType is null && item.ReturnType is not null)
+                {
+                    continue;
+                }
+                if (previousMethod.ReturnType is not null && !previousMethod.ReturnType.EqualsByName(item.ReturnType))
+                {
+                    continue;
+                }
+
+                var parameters = item.Parameters.Except(previousMethod.Parameters, Parameter.TypeAndNameEqualityComparer);
+                if (parameters.All(x => x.IsOptionalInSignature))
+                {
+                    missingParameters = parameters.ToList();
+                    currentMethodToCall = item;
+                    return true;
+                }
+            }
+            missingParameters = null;
+            currentMethodToCall = null;
+            return false;
+        }
+
+        private bool CurrentContainAllPreviousParameters(MethodSignature previousMethod, MethodSignature currentMethod)
+        {
+            var set = currentMethod.Parameters.ToHashSet(Parameter.TypeAndNameEqualityComparer);
+            foreach (var parameter in previousMethod.Parameters)
+            {
+                if (!set.Contains(parameter))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private static MethodChangeset? CompareMethods(IEnumerable<MethodSignature> currentMethods, IEnumerable<MethodSignature>? previousMethods)
+        {
+            if (previousMethods is null)
+            {
+                return null;
+            }
+            var missing = new List<MethodSignature>();
+            var updated = new List<(List<MethodSignature> Current, MethodSignature Previous)>();
+            var set = currentMethods.ToHashSet(MethodSignature.ParameterAndReturnTypeEqualityComparer);
+            var dict = new Dictionary<string, List<MethodSignature>>();
+            foreach (var item in currentMethods)
+            {
+                if (!dict.TryGetValue(item.Name, out var list))
+                {
+                    dict.Add(item.Name, new List<MethodSignature> { item });
+                }
+                else
+                {
+                    list.Add(item);
+                }
+            }
+            foreach (var item in previousMethods)
+            {
+                if (!set.Contains(item))
+                {
+                    if (dict.TryGetValue(item.Name, out var currentOverloadMethods))
+                    {
+                        updated.Add((currentOverloadMethods, item));
+                    }
+                    else
+                    {
+                        missing.Add(item);
+                    }
+                }
+            }
+            return new(missing, updated);
+        }
+
+        public IReadOnlyList<MethodSignature> Methods { get; }
+
+        private IReadOnlyList<MethodSignature> PopulateMethodsFromCompilation(Compilation? compilation)
```

Comment:
I believe this method should be static since it is using in the ctor only.

Reply:
This can't be static, because `PopulateMethods` method within is using instance field `_typeFactory`, it can't be static.